src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        addBottom(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkRangeAdd(index);
+        if (index == 0) {
+            addTop(value);
+        } else if (index == size) {
+            addBottom(value);
+        } else {
+            addMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addBottom(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkRange(index);
+        Node<T> nodeAtIndex = getNode(index);
```

Comment:
chechRange() method is called twice: here and within getNode() method. So I'd recommend removing checkRange call from this method. Let's fix the same problem in all places